### PR TITLE
Release 2.6

### LINKED ISSUE #2880: Release 2.6.0
### Release Type

minor release

### **For all releases** Github Workflow Test Matrix Checkup

- [ ] check the testing workflow ([.github/workflows/test.yaml](/Kong/kubernetes-ingress-controller/blob/main/.github/workflows/test.yaml)) and ensure that all matrix versions ([.github/workflows/e2e.yaml](/Kong/kubernetes-ingress-controller/blob/main/.github/workflows/e2e.yaml) and [.github/workflows/release.yaml](/Kong/kubernetes-ingress-controller/blob/main/.github/workflows/release.yaml))  are up to date for various component releases. If there have been any new releases (major, minor or patch) of those components since the latest version seen in that configuration make sure the new versions get added before proceeding with the release. Remove any versions that are no longer supported by the environment provider.
- [ ] Kubernetes (via [KIND](https://hub.docker.com/r/kindest/node/tags) and the latest image available when creating a new rapid channel cluster from the GKE new cluster wizard)
- [ ] Istio (via [Istio's releases page](https://github.com/istio/istio/releases))

### **For all releases** Create Release Branch

- [ ] ensure that you have up to date copy of `main`: git checkout main; git pull
- [ ] create the release branch for the version (e.g. `release/1.3.1`): `git branch -m release/x.y.z`
- [ ] Make any final adjustments to CHANGELOG.md. Double-check that dates are correct, that link anchors point to the correct header, and that you've included a link to the Github compare link at the end.
- [ ] retrieve the latest license report from FOSSA and save it to LICENSES
- [ ] ensure base manifest versions use the new version (`config/image/enterprise/kustomization.yaml` and `config/image/oss/kustomization.yaml`) and update manifest files: `make manifests`
- [ ] push the branch up to the remote: `git push --set-upstream origin release/x.y.z`

### **For all releases** Create a Release Pull Request

- [ ] Check the [latest test run](https://github.com/Kong/kubernetes-ingress-controller/actions/workflows/e2e.yaml) to confirm that E2E tests are succeeding. If you are backporting features into a non-main branch, run a [targeted E2E job against that branch](https://github.com/Kong/kubernetes-ingress-controller/actions/workflows/e2e_targeted.yaml).
- [ ] Open a PR from your branch to `main`.
- [ ] Once the PR is merged, [initiate a release job](https://github.com/Kong/kubernetes-ingress-controller/actions/workflows/release.yaml). Your tag must use `vX.Y.Z` format. Set `latest` to true if this will be the latest release.
- [ ] CI will validate the requested version, build and push an image, and run tests against the image before finally creating a tag and publishing a release. If tests fail, CI will push the image but not the tag or release. Investigate the failure, correct it as needed, and start a new release job.

### **For major/minor releases only** Update Release documents

- [ ] Create a new branch in the [documentation site repo](https://github.com/Kong/docs.konghq.com).
- [ ] Update articles in the new version as needed.
- [ ] Update `references/version-compatibility.md` to include the new versions (make sure you capture any new Kubernetes/Istio versions that have been tested)
- [ ] Copy `app/_data/docs_nav_kic_OLDVERSION.yml` to `app/_data/docs_nav_kic_NEWVERSION.yml` and update the `release` field to `NEWVERSION`. Add entries for any new articles.
- [ ] Add a section to `app/_data/kong_versions.yml` for your version.
- [ ] Open a PR from your branch.

### Release Troubleshooting

The [Release Troubleshooting guide](https://github.com/Kong/kubernetes-ingress-controller/blob/main/RELEASE.md#release-troubleshooting) covers strategies for dealing with a release that has failed.

### Manual Docker image build

If the "Build and push development images" Github action is not appropriate for your release, or is not operating properly, you can build and push Docker images manually

### Steps

- [ ] Check out your release tag.
- [ ] Run `make container`. Note that you can set the `TAG` environment variable if you need to override the current tag in Makefile.
- [ ] Add additional tags for your container (e.g. `docker tag kong/kubernetes-ingress-controller:1.2.0-alpine kong/kubernetes-ingress-controller:1.2.0; docker tag kong/kubernetes-ingress-controller:1.2.0-alpine kong/kubernetes-ingress-controller:1.2`)
- [ ] Create a temporary token for the `kongbot` user (see 1Password) and log in using it.
- [ ] Push each of your tags (e.g. `docker push kong/kubernetes-ingress-controller:1.2.0-alpine`)

### PR DESCRIPTION
**Pull request by rainest**
**What this PR does / why we need it**:
Releases 2.6.

**Which issue this PR fixes**:

Fix #2880 

**Special notes for your reviewer**:

- [ ] Includes a link to documentation that doesn't exist yet in the changelog: https://github.com/Kong/docs.konghq.com/pull/4444
- [x] 3.0 is not yet available in the [official Docker library](https://hub.docker.com/_/kong). It needs to be available before this can go out. It furthermore needs to run, because currently it's available but doing this:
```
15:03:41-0700 esenin $ docker run --rm kong:3.0 /bin/bash -c "echo aaa" 
Unable to find image 'kong:3.0' locally
3.0: Pulling from library/kong
docker: no matching manifest for linux/amd64 in the manifest list entries.
```
- [x] Double-check the changelog date in case the above cause us to delay past the 12th.

### CODE REVIEW

**Comment by mflendrich** [comment]
this probably should take 2880 [blockers](https://github.com/Kong/kubernetes-ingress-controller/issues/2880#issuecomment-1234485062) into account, no?

**Comment by oooohhoo** [comment]
> 3.0 is not yet available in the [official Docker library](https://hub.docker.com/_/kong). It needs to be available before this can go out. It furthermore needs to run, because currently it's available but doing this:

3.0 is now available

File: CHANGELOG.md
Comment (rainest) [inline CHANGELOG.md:56]:
```suggestion
> Release date: 2022-09-14
```

**Comment by rainest** [comment]
The go-kong item from the issue isn't completely done yet, but https://github.com/Kong/kubernetes-ingress-controller/pull/2932 is that and  is currently all green/will merge once someone approves it. This is otherwise ready for review. The changelog will initially have a dead link, but the doc has gone through its initial review and should be ready soon-ish.

**Review by shaneutt — APPROVED** [review]
Everything in here looks fine, let's just make sure we're getting :heavy_check_mark: on the E2E tests before we actually release as there appears a flake occurred last night we should verify it was in fact a flake (and then fix that soon).

**Pull request merged**
